src/Stratis.Bitcoin.Features.Wallet/WalletSyncManager.cs

```diff
@@ -96,157 +120,257 @@ public void Stop()
         }
 
         /// <inheritdoc />
-        public virtual void ProcessBlock(Block block)
+        public void ProcessBlock(Block block)
         {
-            Guard.NotNull(block, nameof(block));
+            this.logger.LogTrace("()");
+
             this.logger.LogTrace("({0}:'{1}')", nameof(block), block.GetHash());
 
-            ChainedHeader newTip = this.chain.GetBlock(block.GetHash());
-            if (newTip == null)
-            {
-                this.logger.LogTrace("(-)[NEW_TIP_REORG]");
-                return;
-            }
+            this.QueueBlock(block);
+
+            this.logger.LogTrace("(-)");
+        }
+
+        /// <inheritdoc />
+        public virtual void ProcessTransaction(Transaction transaction)
+        {
+            this.logger.LogTrace("()");
+
+            Guard.NotNull(transaction, nameof(transaction));
```

Comment:
`Guard` should be the first thing in the method.
You don't need `this.logger.LogTrace("()");` as `this.logger.LogTrace("({0}:'{1}')", nameof(transaction), transaction.GetHash());` acts as the entry log.

Reply:
Ok, yes I realised this when I added debug logging to fix the wallet sync issue.  I should have removed these then.. thanks.